exonum/src/node/consensus.rs

```diff
@@ -270,7 +270,7 @@ impl NodeHandler {
         }
 
         // Commit propose
-        for (round, block_hash) in self.state.unknown_propose_with_precommits(&hash) {
+        for (round, block_hash) in self.state.take_unknown_propose_with_precommits(&hash) {
```

Comment:
Off-topic, but I'm too curious: does the `commit` call in the cycle mean that several blocks can be committed within the cycle? How does it work? I think a comment explaining this stuff could be very helpful. If I understand correctly, this actually cannot happen because enough precommits can be collected for a single propose (as per the safety property of the consensus algorithm), which begs the question why use a vector in `state.unknown_proposes_with_precommits`.

Reply:
I suppose that only single propose can be committed as block on given height thus this cycle is redundant. 
But in my opinion this issue should be fixed during the consesus code refactoring.